inc/compat/plugins/compat-plugin-woo-delivery.php

```diff
@@ -165,15 +239,15 @@ public function maybe_set_step_incomplete( $is_step_complete ) {
 		$pickup_time_settings = FluidCheckout_Settings::instance()->get_option( 'coderockz_woo_delivery_pickup_settings' );
 
 		// Get field values
-		$order_type = FluidCheckout_Steps::instance()->get_checkout_field_value_from_session( 'coderockz_woo_delivery_delivery_selection_box' );
-		$delivery_date = FluidCheckout_Steps::instance()->get_checkout_field_value_from_session( 'coderockz_woo_delivery_date_field' );
-		$delivery_time = FluidCheckout_Steps::instance()->get_checkout_field_value_from_session( 'coderockz_woo_delivery_time_field' );
-		$pickup_date = FluidCheckout_Steps::instance()->get_checkout_field_value_from_session( 'coderockz_woo_delivery_pickup_date_field' );
-		$pickup_time = FluidCheckout_Steps::instance()->get_checkout_field_value_from_session( 'coderockz_woo_delivery_pickup_time_field' );
+		$order_type = WC()->session->get( 'coderockz_woo_delivery_delivery_selection_box' );
+		$delivery_date = WC()->session->get( 'coderockz_woo_delivery_date_field' );
+		$delivery_time = WC()->session->get( 'coderockz_woo_delivery_time_field' );
+		$pickup_date = WC()->session->get( 'coderockz_woo_delivery_pickup_date_field' );
+		$pickup_time = WC()->session->get( 'coderockz_woo_delivery_pickup_time_field' );
```

Comment:
@vlad-dedikov I read your comment on the Clickup task, but it is still not clear why it did not work with the functions from Fluid Checkout.

1. Was it not working because the Coderockz plugin uses the session keys without the prefix `fc_`?
2. Could you describe the issues you had while trying to use the functions from Fluid Checkout?
3. If using the WooCommerce session directly is strictly needed, we should add a comment on the code for why that is the case so we do not refactor that in the future when we forget about these reasons.

Reply:
@diegoversiani I've search-replaced all of those and it appears to be working... I guess I already removed something that was clearing/resetting session values or it had something to do with JS.